File: src/main/java/be/breina/openrgb/model/Color.java
package be.breina.openrgb.model;

import java.io.IOException;
import java.nio.ByteBuffer;
import lombok.experimental.Delegate;
import be.breina.openrgb.util.LittleEndianInputStream;

public class Color {

    @Delegate
    private java.awt.Color color;

    public Color(java.awt.Color color) {
        this.color = color;
    }

    Color(LittleEndianInputStream input) throws IOException {
        final var red = input.read();
        final var green = input.read();
        final var blue = input.read();
        input.skipNBytes(1);

        color = new java.awt.Color(red, green, blue);
    }

    void setColor(java.awt.Color color) {
        this.color = color;
    }

    @Override
    public String toString() {
        return color.toString();
    }
}

File: src/main/kotlin/be/breina/openrgb/model/Color.kt
package be.breina.openrgb.model

import be.breina.openrgb.util.LittleEndianInputStream

@Suppress("MemberVisibilityCanBePrivate")
class Color(r: Byte, g: Byte, b: Byte) {

    var red: Byte = r
        internal set

    var green: Byte = g
        internal set

    var blue: Byte = b
        internal set

    internal companion object {
        internal fun from(input: LittleEndianInputStream): Color {
            val red = input.read().toByte()
            val green = input.read().toByte()
            val blue = input.read().toByte()
            input.skipNBytes(1)

            return Color(red, green, blue)
        }
    }
}